src/FunderMaps.Core.Authentication/SignInService.cs

```
@@ -0,0 +1,226 @@
+        }
+
+        /// <summary>
+        ///     Attempts to sign in the specified <paramref name="user"/> and <paramref name="password"/> combination.
+        /// </summary>
+        /// <param name="user">The user to sign in.</param>
+        /// <returns>Instance of <see cref="SignInContext"/>.</returns>
+        public virtual async Task<SignInContext> SignInAsync(IUser user, ITenant tenant, OrganizationRole organizationRole, string authenticationType)
+        {
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+
+            if (tenant == null)
+            {
+                throw new ArgumentNullException(nameof(tenant));
+            }
+
+            var result = await CanSignInAsync(user.Id);
+            if (result != SignInContext.Success)
+            {
+                return result;
+            }
+
+            Logger.LogInformation(1, $"User {user} sign in was successful.");
+
+            return new SignInContext(
+                result: AuthResult.Success,
+                principal: PrincipalProvider.CreateTenantUserPrincipal(user, tenant, organizationRole, authenticationType));
+        }
+
+        /// <summary>
+        ///     Attempts to sign in the specified <paramref name="userName"/> and <paramref name="password"/> combination.
+        /// </summary>
+        /// <param name="email">The user email to sign in.</param>
+        /// <param name="password">The password to attempt to sign in with.</param>
+        /// <returns>Instance of <see cref="SignInContext"/>.</returns>
+        public virtual async Task<SignInContext> PasswordSignInAsync(string email, string password, string authenticationType)
+        {
+            IUser user = await UserRepository.GetByEmailAsync(email);
+            if (user == null)
+            {
+                return SignInContext.Failed;
```

Maybe throw a custom exception here?